tachyon_core/src/storage/file.rs

```
@@ -388,6 +465,7 @@ impl CursorImpl {
         self.use_query_hint_for_value(self.value);
 
         if self.current_timestamp > self.end {
+            self.file_lock.release().unwrap();
```

TODO: we can ignore error here potentially

---

TODO: we should probably flush pages here too

---

(added flush)